test/t/test_rsync.py

```diff
@@ -18,3 +22,50 @@ def test_3(self, completion):
     @pytest.mark.complete("rsync --", require_cmd=True)
     def test_4(self, completion):
         assert "--help" in completion
+
+    def test_vercomp(self, bash):
+        def _test_vercomp(ver1, ver2, result):
+            output = assert_bash_exec(
+                bash,
+                f"_comp_cmd_rsync__vercomp {ver1} {ver2}; echo $?",
+                want_output=True,
+            ).strip()
+            if result == "=":
+                assert output == "0"
+            elif result == ">":
+                assert output == "1"
+            elif result == "<":
+                assert output == "2"
+            else:
+                raise Exception(f"Unsupported comparison result: {result}")
+
+        _test_vercomp("1", "1", "=")
+        _test_vercomp("1", "2", "<")
+        _test_vercomp("2", "1", ">")
+        _test_vercomp("1.1", "1.2", "<")
+        _test_vercomp("1.2", "1.1", ">")
+        _test_vercomp("1.1", "1.1.1", "<")
+        _test_vercomp("1.1.1", "1.1", ">")
+        _test_vercomp("1.1.1", "1.1.1", "=")
+        _test_vercomp("2.1", "2.2", "<")
+        _test_vercomp("3.0.4.10", "3.0.4.2", ">")
+        _test_vercomp("4.08", "4.08.01", "<")
+        _test_vercomp("3.2.1.9.8144", "3.2", ">")
+        _test_vercomp("3.2", "3.2.1.9.8144", "<")
+        _test_vercomp("1.2", "2.1", "<")
+        _test_vercomp("2.1", "1.2", ">")
+        _test_vercomp("5.6.7", "5.6.7", "=")
+        _test_vercomp("1.01.1", "1.1.1", "=")
+        _test_vercomp("1.1.1", "1.01.1", "=")
+        _test_vercomp("1", "1.0", "=")
+        _test_vercomp("1.0", "1", "=")
+        _test_vercomp("1.0.2.0", "1.0.2", "=")
+        _test_vercomp("1..0", "1.0", "=")
+        _test_vercomp("1.0", "1..0", "=")
```

Comment:
Can you use `@pytest.mark.parametrize`? I haven't tested it but something like this:

```suggestion
    @pytest.mark.parametrize(
        "ver1,ver2,result",
        [
            ("1", "1", "="),
            ("1", "2", "<"),
            ("2", "1", ">"),
            ("1.1", "1.2", "<"),
            ("1.2", "1.1", ">"),
            ("1.1", "1.1.1", "<"),
            ("1.1.1", "1.1", ">"),
            ("1.1.1", "1.1.1", "="),
            ("2.1", "2.2", "<"),
            ("3.0.4.10", "3.0.4.2", ">"),
            ("4.08", "4.08.01", "<"),
            ("3.2.1.9.8144", "3.2", ">"),
            ("3.2", "3.2.1.9.8144", "<"),
            ("1.2", "2.1", "<"),
            ("2.1", "1.2", ">"),
            ("5.6.7", "5.6.7", "="),
            ("1.01.1", "1.1.1", "="),
            ("1.1.1", "1.01.1", "="),
            ("1", "1.0", "="),
            ("1.0", "1", "="),
            ("1.0.2.0", "1.0.2", "="),
            ("1..0", "1.0", "="),
            ("1.0", "1..0", "="),
        ]
    )
    def test_vercomp(self, bash, ver1, ver2, result):
        output = assert_bash_exec(
            bash,
            f"_comp_cmd_rsync__vercomp {ver1} {ver2}; echo $?",
            want_output=True,
        ).strip()
        if result == "=":
            assert output == "0"
        elif result == ">":
            assert output == "1"
        elif result == "<":
            assert output == "2"
        else:
            raise Exception(f"Unsupported comparison result: {result}")
```